### DISCUSSION TITLE
Borg backup: files found but no archive located

### DISCUSSION THREAD

**Discussion by oms9** [nextcloud/all-in-one#2822] [Questions] [answered]
### Discussed in https://github.com/nextcloud/all-in-one/discussions/2571:

Hello, this is still unresolved and I do not know how to proceed with handling this problem. My best guess from my numerous failed attempts is that the backup container is unable to view files or directories in Windows 11.

<div type='discussions-op-text'>

<sup>Originally posted by **BustedSec** May 19, 2023</sup>
I am trying to migrate a nextcloud-aio instance from linode to local storage. I created a VM and installed nextcloud-aio, but I'm having an issue locating the borg backup files. The job runs fine and creates the borg folder, but within that folder there is no archive by the name that is given in the backup log.  I tried rsyncing the entire folder over to the new machine and restoring from it, but it doesn't recognize it as a borg backup

The file structure is as follows: 
config  data  hints.74  index.74  integrity.74  nonce  README

here is the log:
```
initializing repository...
using builtin fallback logging configuration
33 self tests completed in 0.23 seconds
Initializing repository at "/mnt/borgbackup/borg"
Key in "<Repository /mnt/borgbackup/borg>" created.
Keep this key safe. Your data will be inaccessible without it.
check_free_space: few segments, not requiring a full free segment
check_free_space: calculated working space for compact as 0 bytes
check_free_space: required bytes 169138, free bytes 79643910144
security: previous location file /root/.config/borg/security/0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304/location not found
security: manifest timestamp file /root/.config/borg/security/0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304/manifest-timestamp not found
security: determined newest manifest timestamp as 
security: remembering previously unknown repository
security: saving state for 0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304 to /root/.config/borg/security/0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304
security: current location   /mnt/borgbackup/borg
security: key type           5
security: manifest timestamp 2023-05-20T00:56:29.881384
security: repository checks ok, allowing access
Synchronizing chunks cache...
Archives: 0, w/ cached Idx: 0, w/ outdated Idx: 0, w/o cached Idx: 0.
Verified integrity of /mnt/borgbackup/borg/index.1
Done.
security: saving state for 0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304 to /root/.config/borg/security/0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304
security: current location   /mnt/borgbackup/borg
security: key type           5
security: manifest timestamp 2023-05-20T00:56:29.881384

By default repositories initialized with this version will produce security
errors if written to with an older version (up to and including Borg 1.0.8).

If you want to use these older versions, you can disable the check by running:
borg upgrade --disable-tam /mnt/borgbackup/borg

See https://borgbackup.readthedocs.io/en/stable/changes.html#pre-1-0-9-manifest-spoofing-vulnerability for details about the security implications.

IMPORTANT: you will need both KEY AND PASSPHRASE to access this repo!
If you used a repokey mode, the key is stored in the repo, but you should back it up separately.
Use "borg key export" to export the key, optionally in printable format.
Write down the passphrase. Store both at safe place(s).

Repository successfully initialized.
Performing backup...
Starting the backup...
Creating archive at "/mnt/borgbackup/borg::20230520_005631-nextcloud-aio"
------------------------------------------------------------------------------
Repository: /mnt/borgbackup/borg
Archive name: 20230520_005631-nextcloud-aio
Archive fingerprint: 7f70d71f5ec4ab66432bfdb517f791862f019b7e9d30743a6a7068ad52a7f551
Time (start): Sat, 2023-05-20 00:56:31
Time (end):   Sat, 2023-05-20 01:05:22
Duration: 8 minutes 51.23 seconds
Number of files: 74573
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:               39.22 GB             35.90 GB             34.28 GB
All archives:               39.22 GB             35.90 GB             34.28 GB

                       Unique chunks         Total chunks
Chunk index:                   77851                88536
------------------------------------------------------------------------------
Pruning the archives...
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
Deleted data:                    0 B                  0 B                  0 B
All archives:               39.22 GB             35.90 GB             34.28 GB

                       Unique chunks         Total chunks
Chunk index:                   77851                88536
------------------------------------------------------------------------------
Compacting the archives...
Backup finished successfully on 20.05.2023 - 01:05:26 (00 hours 08 minutes 55 seconds)
```
</div>

**Comment by szaimen** [comment, marked as answer]
Hi, can you please add some steps what you tried to do exactly and what didnt work?

**Reply by oms9** [reply]
Hello, I created a backup from using my AIO VirtualMachine on Virtualbox. I want to migrate it to Docker running directly on Windows 11, bypassing Virtualbox. I copied the backup from the VM using a shared folder and placed it on my D: drive on Windows.

I then installed the AIO in Docker using the instructions and it launched just fine, I then clicked restore in the AIO web-interface and pointed it to the directory where the backup lives but it was unable to detect the backup, and in the error messages where its supposed to list the directories and files present where it looked, nothing was listed other than current and parent (`.` and `..`).

There are 2 error messages I have grown accustomed to, the first one is:
```
No 'borg' directory in the given backup directory found!
Only the files/folders below have been found in the given directory.
.
..
Please adjust the directory so that the borg archive is positioned in a folder named 'borg' inside the given directory!
```
While the second is after messing with the path provided in the web-interface: 
```
A 'borg' directory was found but could not find the borg archive.
Only the files/folders below have been found in the borg directory.
.
..
The archive and most importantly the config file must be positioned directly in the 'borg' subfolder.```

I have tried pretty much every way to write the path to where the backup is located to no-avail.

**Reply by szaimen** [reply]
Which directory did you enter in the AIO interface and in which folder exactly is the archive located? Can you post a screenshot of the archive location opened in your windows explorer?

**Reply by oms9** [reply]
Sure!

Path used: `/run/desktop/mnt/host/d/Nextcloud/Backups/backup`
as well as: `/run/desktop/mnt/host/d/nextcloud/backups/backup`

![image](https://github.com/nextcloud/all-in-one/assets/102495444/e7526155-c8ff-4be0-92c5-741fe4f8a387)

**Reply by oms9** [reply]
Forgot to click the address bar, my bad:

![image](https://github.com/nextcloud/all-in-one/assets/102495444/80497bf2-5d80-4445-9e3f-e546bbf31152)

**Reply by szaimen** [reply]
Is this a network drive or a local drive?

**Reply by oms9** [reply]
Local drive

**Reply by szaimen** [reply]
I see. Can you try to restart the docker daemon and try again?

**Reply by oms9** [reply]
Sure, is there a specific command I should use?

**Reply by szaimen** [reply]
No, simply restart docker desktop
![image](https://github.com/nextcloud/all-in-one/assets/42591237/7326775d-6041-4a11-a4eb-9b167249a69c)

**Reply by oms9** [reply]
Still, same error message after a restart and even an AIO master container update.

**Reply by szaimen** [reply]
Hm... is this setting enabled in docker desktop?
![image](https://github.com/nextcloud/all-in-one/assets/42591237/d96b8de9-f09a-4a6c-8e12-c4f7ee4fbd55)

**Reply by oms9** [reply]
Yup, enabled and directed at an Ubuntu installation

**Reply by szaimen** [reply]
Hm... what kind of file system does the drive has? NTFS or FAT?

**Reply by oms9** [reply]
NTFS, it is a 3TB HDD connected over SATA.

**Reply by szaimen** [reply]
All right. Can you check if it works if you move the borg archive for a test back to somewhere on the C drive?

**Reply by szaimen** [reply]
Another test that you could do would be creating a backup dir volume:
```
docker volume create ^
--driver local ^
--name nextcloud_aio_backupdir ^
-o device="/host_mnt/d/Nextcloud/Backups/backup" ^
-o type="none" ^
-o o="bind"
```

And then entering `nextcloud_aio_backupdir` in the aio interface as location to restore from. Does that make it work?

**Reply by oms9** [reply]
I went directly to the docker volume create approach and it worked, thank you so much!

**Reply by oms9** [reply]
Spoke a little too soon, I got another error message about running out of space. (I have the drive space available)

What should I increase? I am a little confused

```
nextcloud_aio_nextcloud_data/admin/files/InstantUpload/Camera/PXL_20221221_193614683.MP.jpg
rsync: [receiver] write failed on "/nextcloud_aio_volumes/nextcloud_aio_nextcloud_data/admin/files/InstantUpload/Camera/PXL_20221221_193210718.jpg": No space left on device (28)
rsync error: error in file IO (code 11) at receiver.c(380) [receiver=3.2.7]
rsync: [sender] write error: Broken pipe (32)
Something failed while restoring from backup.
sending incremental file list
delta-transmission disabled for local transfer or --whole-file
configuration.json
total: matches=0  hash_hits=0  false_alarms=0 data=1243

sent 1.36K bytes  received 102 bytes  2.92K bytes/sec
total size is 1.24K  speedup is 0.85
```

**Reply by szaimen** [reply]
I see, did you configure https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir?

**Reply by oms9** [reply]
Yeah, my run command is as follows:

`sudo docker run --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --publish 80:80 --publish 8080:8080 --publish 8443:8443 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro --env NEXTCLOUD_MOUNT="/run/desktop/mnt/host/d/NextCloud/Backups" --env NEXTCLOUD_DATADIR="/run/desktop/mnt/host/d/NextCloud/Data" nextcloud/all-in-one:latest`

**Reply by szaimen** [reply]
Maybe then actually `-e NEXTCLOUD_DATADIR="/run/desktop/mnt/host/d` is restoring files into the wsl drive. Maybe you need to go the docker volume create route there as well...
